packages/sync/src/class-listener.php

```diff
@@ -353,6 +356,28 @@ public function enqueue_action( $current_filter, $args, $queue ) {
 		}
 	}
 
+	/**
+	 * Sync Data Loss Handler
+	 *
+	 * @param Queue $queue Sync queue.
+	 * @return boolean was send successful
+	 */
+	public function sync_data_loss( $queue ) {
+		if ( ! Settings::is_sync_enabled() ) {
+			return;
+		}
+		Health::update_status( Health::STATUS_OUT_OF_SYNC );
```

Comment:
Just thought of an edge case. This will give us a large number of 'jetpack_sync_data_loss' events potentially if the site keeps trying to enqueue. We should check if status is already out_of_sync. If so then don't update or send the action.

basically add an || Health::STATUS_OUT_OF_SYNC === Health::get_status() to line 366